File: server-http-handler/nodejs/thumbnailator/handler.js
const sharp = require('sharp')
const fs = require('fs')
const scale = parseFloat(process.env.scale)
const image_path = process.env.image_path
const image = sharp(image_path)

function resizer(scale, image) {
	return image
		.metadata()
		.then((metadata) => {
			return image
				.resize(metadata.width * scale, metadata.height * scale)
				.toBuffer()
				//.toFile('output.jpg')
		})
}

function handle(req, res) {
	return resizer(scale, image)
		.then(() => {
			res.writeHead(200);
			return res
		})
		.catch((error) => {
			res.writeHead(500);
			res.write(error)
			return res
		})
}

module.exports.handle = handle;

File: server-http-handler/python/thumbnailator/handler.py
from utils.http_entities import Response
from PIL import Image
import os

scale = float(os.environ['scale'])
image_path = os.environ['image_path']
image = Image.open(image_path)

def handle(request):
	response = Response()
	try:
		image.resize((int(image.width * scale), int(image.height * scale)), Image.BILINEAR)
		if request.get_header('x-save-image'):
			image.save('output.jpg')
		response.set_status_code(200)
	except Exception as e:
		response.set_status_code(500)
		response.set_body(str(e) + '\n')

	return response